### DISCUSSION TITLE
Specifying global document overrides for tests?

### DISCUSSION THREAD

**Discussion by johnrengelman** [open-policy-agent/community#460] [OPA and Rego] [answered]
I have a situation where I have a project that generates a bundle and in that bundle I want to include a skeleton data document (using `data.json` file in the root) that will be patched at runtime by a bundle service. The difficulty I'm running into is what to do with the local test fixtures.
I can't put them into `data.json` as they would be bundled in and I want to avoid that. 
Specifying them in each test rule is cumbersome and would be very repetitious. So that leaves me with 2 options:
1) Put the test fixtures into directories (eg `applications/data.json`) that map to each top-level document. Downside, is that I have to make sure if any developer adds a top-level document, that they update the `opa build` command to add `--ignore <path>` so that it isn't included in the bundle.
2) Put the data into a single folder (eg `fixtures/data.json`). Downside is that each test rule has to specify a `with data.applications as data.fixtures.applications` which is very repetitious.

Ideally, if I could specify a global document override to the `opa test` command, I could avoid this situation Or, better yet, if there was a _special_ path that was always automatically remapped only under `opa test`. So `fixtures/applications/data.json` would be automatically mapped to `data.applications` instead of `data.fixtures.applications`.

(coming from a java background, this is similar to how `src/test/resources` will override anything in `src/main/resources` in standard project formats.

Would a feature like this be something that would be considered? Any other alternatives?

**Comment by ashutosh-narkar** [comment]
Maybe I'm missing something but what is the need to include tests in your final bundle? I would imagine you test all your rules and then build the bundle w/o tests in it.

**Comment by anderseknert** [comment, marked as answer]
> Put the data into a single folder (eg fixtures/data.json). Downside is that each test rule has to specify a with data.applications as data.fixtures.applications which is very repetitious.

Not sure you'd need that. If you run the tests without the `--bundle` flag, like:

```
opa test bundle/ fixtures/
```
you could just have a `data.json` file in the `fixtures` dir where the path was provided in the file rather than by directory structure, e.g. 
```json
{
  "applications": {
    "...": "..."
  }
}
```
```

**Reply by johnrengelman** [reply]
Ah, that's what I was missing. Specifying an additional path, puts it on the root of the tests. 
Thanks @anderseknert
